specs/pvs/corla.pvs

```diff
@@ -1650,6 +1772,186 @@ END system_assumptions
 
 %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
 
+%%% \section{System Logging}
+
+%%% The system must log a variety of events. One reason to log
+%%% information is to help understand how the system is operating, fix
+%%% bugs post-facto, understand how users are using the system,
+%%% etc. Another, more important reason to log in this context of this
+%%% system is to provide an indeliable record of administrator and
+%%% auditor actions so that any audit can be "replayed" by any third
+%%% party.
+
+logging: THEORY
+BEGIN
+  log: TYPE = sequence[string]
+  chain_of_custody_log: TYPE FROM log
+% @trace Per email from Jerome Lovato at CDOS on 19 July 2017: Logging
+% of Audit Board and SOS user errors during use of the system [...is
+% the only logging we care about].
+  rla_tool_log: TYPE FROM log
+END logging
+
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+
+%%% \section{Data Model}
```

Comment:
For reference, here's my Django code for modeling the database in `audit_cvrs`, which I used to track CVR selections, status and discrepancies in several of the Colorado pilot RLAs. Note it doesn't (yet) have logic to store contest selections in the database or automatically compare the selected CVR to the auditors interpretation of the paper ballot.
https://github.com/nealmcb/audit_cvrs/blob/master/audit_cvrs/models.py